### ISSUE TITLE
Confusion about the custom-implemented CE

### ISSUE THREAD

**Issue by blueardour** [ChenJoya/objnessdet#9]
Hi,

The work is really interesting and provides me new thought in my model compression work. However, I felt somesort confused about the cross entrpy developed in the repo.

As far as I could understand, for the cls branch, Focal loss is leveraged in original RetinaNet / FCOS, which outputs multi-label classification score. The categories are no mutually exclusive (sum of the possibility is not 1). In respect to the developed CE, it is still multi-label classification loss function.

Might I ask

1. Is it possible to employ nn.CROSSENTROPYLOSS for the classification branch instead?  Namely multi-category vs multi-label. Indeed, I saw someone mentioned that objects subject to different categories might overlap in the same positision. It is more suitable to use multi-label rather than multi-category loss. But I observed that most objects are not so crowed (or even it is crowed, the overlapped objects generally belong to the same category).

2. Is it equivalent by setting gamma to be zero in Focal loss compared to the develpoped CE module?  If so, it might be not necessary to re-implement the CE module?

**Comment by chenjoya** [comment]
Hi, sorry for the late response:

1. I guess you can try the softmax loss (nn.CROSSENTROPYLOSS). But in fact, our Sampling-Free and Focal Loss are designed for binary-loss of multi-class classification. Softmax loss with Sampling-Free may need another form for initialization.

2. You can set gamma=0, alpha=0.5, and then multiply the loss * 2, which is equivalent to binary ce loss with sigmoid. We re-implement for the faster speed.

**Comment by blueardour** [comment]
Hi, @ChenJoya 

Thanks for the reply. I will have a try on myself about the chosen of classifcation loss function.
